src/dialects/postgres/query.js

```diff
@@ -85,7 +85,11 @@ class Query extends AbstractQuery {
       queryResult = await query;
     } catch (error) {
       // set the client so that it will be reaped if the connection resets while executing
-      if (error.code === 'ECONNRESET') {
+      if (error.code === 'ECONNRESET'
+        || new RegExp(/SSL SYSCALL error: EOF detected/i).test(error)
```

Comment:
There is no need for `new RegExp` since `/SSL SYSCALL error: EOF detected/i` is a RegExp object already

Reply:
Thank you, corrected.